src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -419,9 +419,10 @@
                     function(xhr, el) {
 
                         /* Evaluate in global scope scripts embedded inside the toolbar */
-                        eval.call({}, ([].slice.call(el.querySelectorAll('script')).map(function (script) {
-                          return script.firstChild.nodeValue;
-                        }).join(';\n')));
+                        var i, scripts = [].slice.call(el.querySelectorAll('script'));
+                        for (i = 0; i < scripts.length; ++i) {
```

Comment:
Why not define `i` inside the loop? `for (var i...`, that way you can be sure you won't get any scope bleeding.

Reply:
Because this variable is not only used by this loop but also by next ones. Putting it inside the loop might make us remove the variable declaration by mistake in a refactoring, which would make us use a global variable here.

> that way you can be sure you won't get any scope bleeding.

That's wrong. It would be true when using `let`, but not for `var` (which is scoped only by functions, not by loops)

Reply:
I was confusing it with `let` indeed. In case of `var` this is indeed better :+1: